dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/MultipleApproverActionlet.java

```diff
@@ -64,95 +54,93 @@ public void executeAction(WorkflowProcessor processor, Map<String, WorkflowActio
+					Logger.error(this.getClass(), "Unable to find user with userID:" + userIdToken);
 				}
 
 			}
 		}
+
 		List<WorkflowHistory> histories = processor.getHistory();
 		
 		// add this approval to the history
-		WorkflowHistory h = new WorkflowHistory();
-		h.setActionId(processor.getAction().getId());
-		h.setMadeBy(processor.getUser().getUserId());
-		if(histories == null){
-			histories = new ArrayList<WorkflowHistory>();
-			histories.add(h);
-		}else histories.add(h);
+		final WorkflowHistory workflowHistory = new WorkflowHistory();
+		workflowHistory.setActionId(processor.getAction().getId());
+		workflowHistory.setMadeBy(processor.getUser().getUserId());
+		histories = histories == null?new ArrayList<>():histories;
+		histories.add(workflowHistory);
 		
-		for (User u : requiredApprovers) {
+		for (final User requiredApprover : requiredApprovers) {
+			for (final WorkflowHistory history : histories) {
 
-			for (WorkflowHistory history : histories) {
-				if (history.getActionId().equals(processor.getAction().getId())) {
-					if (u.getUserId().equals(history.getMadeBy())) {
-						hasApproved.add(u);
-					}
+				final Map<String, Object> changeMap = history.getChangeMap();
+				if (history.getActionId().equals(processor.getAction().getId()) && // if it is the action id and it is not reset.
+						!WorkflowHistoryState.RESET.name().equals(changeMap.get("state"))) {
 
-				}
+					if (requiredApprover.getUserId().equals(history.getMadeBy())) {
 
+						hasApproved.add(requiredApprover);
+					}
+				}
 			}
-
 		}
 		
 		if (hasApproved.size() < requiredApprovers.size()) {
 			
 			shouldStop = true;
 			// keep the workflow process on the same step
 			processor.setNextStep( processor.getStep());
-			
-			
+
 			// only send emails to users who have not approved
-			List<String> emails = new ArrayList<String>();
-			for (User u : requiredApprovers) {
-				if(!hasApproved.contains(u)){
-					emails.add(u.getEmailAddress());					
+			final List<String> emails = new ArrayList<>();
+			for (final User user : requiredApprovers) {
+				if(!hasApproved.contains(user)){
+					emails.add(user.getEmailAddress());
 				}
 			}
 			
 			// to assign it for next assignee
-			for (User u : requiredApprovers) {
-				if(!hasApproved.contains(u)){					
+			for (final User requiredApprover : requiredApprovers) {
+				if(!hasApproved.contains(requiredApprover)){
 					try {
-	                   processor.setNextAssign(APILocator.getRoleAPI().getUserRole(u));
+	                   processor.setNextAssign(APILocator.getRoleAPI().getUserRole(requiredApprover));
 	                   break;
 	                } catch (DotDataException e) {
 	                   Logger.error(MultipleApproverActionlet.class,e.getMessage(),e);
 	                }
 				}
 			}
-			
-			
-			
-			
-			String[] emailsToSend = (String[]) emails.toArray(new String[emails.size()]);
 
-			
+			final String[] emailsToSend = emails.toArray(new String[emails.size()]);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This call to Collection.toArray() may be optimizable](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4904110)